app/search/models.py

```diff
@@ -43,6 +43,43 @@ def get_latest_test_results():
 
     return _get_latest_test_results(normalized_date)
 
+class DatasetCache(object):
+    def __init__(self, current_app):
+        self.current_app = current_app
+        dataset_cache_dir = current_app.config['DATASET_CACHE_PATH']
+        if not os.path.exists(dataset_cache_dir):
+            os.makedirs(dataset_cache_dir)
+
+    @property
+    def cachedDatasets(self):
+        return dict(
+            (f.name, f)
+            for f in os.scandir(self.current_app.config['DATASET_CACHE_PATH'])
+        )
+
+    def getZippedContent(self, dataset):
+        """
+          1. Server checks if a zip file already exists for this version.
+          2. Return zip file
+        """
+
+        datasetmeta = DATSDataset(dataset.fspath)
+        zipFilename = '.'.join([
+            datasetmeta.name.replace('/','__'),
+            datasetmeta.version,
+           'tar',
+           'gz'
+        ])
+
+        cached = self.cachedDatasets.get(zipFilename)
+        zipped = cached.path if cached is not None else None
+
+        if zipped is None:
+            raise RuntimeError('Not found')
+
+        os.system('touch -a ' + zipped)
```

Comment:
not sure I understand this line. Was it for testing purposes and it should be removed?

Reply:
This is updating the 'last accessed timestamp' of the file on the filesystem.

Reply:
It was used when I was deleting the least recently accessed. I forgot to remove the line.